language-support/ts/daml-ledger/index.ts

```diff
@@ -500,21 +502,11 @@ class Ledger {
       const json: unknown = JSON.parse(event.data.toString());
       if (isRecordWith('events', json)) {
         const events = jtv.Result.withException(jtv.array(decodeEvent(template)).run(json.events));
-        if (events.length == 0 && isRecordWith('offset', json)) {
-          if (!haveSeenEvents) {
-            // NOTE(MH): If we receive the marker indicating that we are switching
-            // from the ACS to the live stream and we haven't received any events
-            // yet, we signal this by pretending we received an empty list of
-            // events. This never does any harm.
-            haveSeenEvents = true;
-            emitter.emit('change', state, []);
-          }
-        } else {
-          state = change(state, events);
-          if (isRecordWith('offset', json)) {
-            haveSeenEvents = true;
-          }
-          emitter.emit('change', state, events);
+        state = change(state, events);
+        emitter.emit('change', state, events);
```

Comment:
We don't want to trigger any changes if there were no events. This happens as part of heart beating and informing about new ledger offsets.
```suggestion
        if (events.length > 0) {
          state = change(state, events);
          emitter.emit('change', state, events);
        }
```